Update elasticsearch-keystore.md with special character handling and echo command to enter the password. (#127135)
elastic/elasticsearch: docs/reference/elasticsearch/command-line-tools/elasticsearch-keystore.md

`-v, --verbose`
:   Shows verbose output.

## Handling special characters

> 

Reviewer: one final option if we want an admonition here :)

> 

Author: yes that would be good.

`-v, --verbose`
:   Shows verbose output.

## Handling special characters

:::{important}
Improper handling of special characters can lead to authentication failures and service outages.
:::

> Exclamation mark: `!`
> : When using the shell, the combination `\!` is stored as just `!`. This can lead to authentication failures if the password is not stored as intended.
> 
> Quotes: `"`
> : If quotes are used around the password, they become part of the password itself. This can cause the password to be incorrect when retrieved from the keystore.
> 
> Backslash: `\`
> : The backslash character needs to be properly escaped. If not escaped correctly, it may be omitted or misinterpreted, leading to incorrect password storage.

Reviewer: ```suggestion
**Exclamation mark**: `!`
: When using the shell, the combination `\!` is stored as just `!`. This can lead to authentication failures if the password is not stored as intended.

**Quotes**: `"`
: If quotes are used around the password, they become part of the password itself. This can cause the password to be incorrect when retrieved from the keystore.

**Backslash**: `\`
: The backslash character needs to be properly escaped. If not escaped correctly, it may be omitted or misinterpreted, leading to incorrect password storage.
```

Reviewer: last cosmetic edit I promise :)

```sh
cat /file/containing/setting/value | bin/elasticsearch-keystore add --stdin the.setting.name.to.set
```

> Alternatively you can use the echo command to input the password into the keystore. For example if the password is `your\!password` :

Reviewer: ```suggestion
Alternatively, you can use the echo command to input the password into the keystore. For example, if the password is `your\!password` :
```

```sh
echo -n 'your\!password' | bin/elasticsearch-keystore add the.setting.name.to.set
```

Values for multiple settings must be separated by carriage returns or newlines.

> ### Common Issues with Special Characters [special-chracter-issues]
> Exclamation Mark (`\!`): When using the shell, the combination `\!` is stored as just `!`. This can lead to authentication failures if the password is not stored as intended.
> 
> Quotes: If quotes are used around the password, they are stored as part of the password. This can cause the password to be incorrect when retrieved from the keystore.
> 
> Backslash (`\`): The backslash character needs to be escaped properly. If not escaped, it may be omitted or misinterpreted, leading to incorrect password storage.

Reviewer: ```suggestion
## Handling special characters

Exclamation Mark `!`
: When using the shell, the combination `\!` is stored as just `!`. This can lead to authentication failures if the password is not stored as intended.

Quotes `"`
: If quotes are used around the password, they become part of the password itself. This can cause the password to be incorrect when retrieved from the keystore.

Backslash `\`
: The backslash character needs to be properly escaped. If not escaped correctly, it may be omitted or misinterpreted, leading to incorrect password storage.
```

Reviewer: On top of changing this whole section to be a H2, and moving it before the examples, I'd recommend using [definition list](https://elastic.github.io/docs-builder/syntax/definition-lists/) here.

Reviewer: This also helps future proof the page structure, this could be nested under a more general **Troubleshooting** heading if we need to update this page with more guidance over time
